File: elixir/line-up/1/lib/line_up.ex
defmodule LineUp do
  defp ending_number(number) do
    cond do
      String.ends_with?(number, "1") and not String.ends_with?(number, "11") -> "st"
      String.ends_with?(number, "2") and not String.ends_with?(number, "12") -> "nd"
      String.ends_with?(number, "3") and not String.ends_with?(number, "13") -> "rd"
      true -> "th"
    end
  end

  @doc """
  Formats a full ticket sentence for the given name and number, including
  the person's name, the ordinal form of the number, and fixed descriptive text.
  """
  @spec format(name :: String.t(), number :: pos_integer()) :: String.t()
  def format(name, number) do
    "#{name}, you are the #{Integer.to_string(number) <> ending_number(Integer.to_string(number))} customer we serve today. Thank you!"
  end
end

File: python/line-up/1/line_up.py
def line_up(name, number):
    number_str = str(number)
    def choose_suffix():
        if number_str.endswith("1") and not number_str.endswith("11"):
            return "st"
        if number_str.endswith("2") and not number_str.endswith("12"):
            return "nd"
        if number_str.endswith("3") and not number_str.endswith("13"):
            return "rd"
        return "th"

    return f"{name}, you are the {number_str + choose_suffix()} customer we serve today. Thank you!"
